agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/BackCompatMetricData.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.exporter;
+
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.metrics.data.Data;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+class BackCompatMetricData implements MetricData {
+
+  private static final Pattern NAME_AND_TAG_KEY_PATTERN = Pattern.compile("[^a-zA-Z0-9\\-]");
+
+  private final MetricData delegate;
+  private final String name;
+
+  BackCompatMetricData(MetricData delegate) {
+    this.delegate = delegate;
+    this.name = backCompatName(delegate.getName());
+  }
+
+  private static String backCompatName(String name) {
+    return NAME_AND_TAG_KEY_PATTERN.matcher(toSnakeCase(name)).replaceAll("_");
+  }
+
+  private static String toSnakeCase(String value) {
```

Comment:
what does it mean `toSnakeCase`?

Reply:
https://en.wikipedia.org/wiki/Snake_case